robottelo/helpers.py

```diff
@@ -716,3 +716,64 @@ def idgen(val):
     test name
     """
     return '_parameter'
+
+
+class InstallerCommand:
+    """This class constructs, parses, updates and gets formatted installer commands"""
+
+    def __init__(self, *args, command='satellite-installer', allow_dupes=False, **kwargs):
+        self.command = command
+        self.args = kwargs.pop('installer_args', [])
+        self.opts = kwargs.pop('installer_opts', {})
+        self.update(*args, allow_dupes=allow_dupes, **kwargs)
+
+    def get_command(self):
+        command_str = self.command
+        for arg in self.args:
+            command_str += f' {"-" if len(arg) == 1 else "--"}{arg}'
+        for key, val in self.opts.items():
+            if isinstance(val, list):
+                for v in val:
+                    command_str += f' --{key.replace("_", "-")} {v}'
+            else:
+                command_str += f' --{key.replace("_", "-")} {val}'
+        return command_str
+
+    def update(self, *args, allow_dupes=False, **kwargs):
+        new_args = [arg for arg in args if arg not in self.args]
+        self.args.extend(new_args)
+        if not allow_dupes:
```

Comment:
is this `if not allow_dupes:` is redundant ? or necessary two places where it is working same object similarly

Reply:
allow_dupes is here so people can either add an additional value for the same option or override an existing one.